libflux/fluxdoc/SPEC.md

```diff
@@ -0,0 +1,284 @@
+# Writing Flux Doc Comments
+
+Flux will treat any uninterrupted series of line comments immediately preceding a function definition, package clause, builtin statement, or any declaration of an option or constant, as a doc comment.
+
+A flux doc comment consists of multiple sections, which are defined in this document. Each section should be separated from other sections by a single blank line comment. Unless otherwise noted, each section is required, and the sections must be arranged in the order in which they are presented in this document.
+
+## Documenting functions
+
+### Name and Description
+
+Doc comments for functions should begin with the function name, wrapped in backticks, followed by a brief description of what the function does or how it is used. The name and description should read as one or more full sentences.
+
+_Example:_
+
+```
+// `join` merges two input streams into a single output stream based on columns
+// with equal values.
+```
+
+### Extra Details
+
+Some function documentation may require explanation beyond what is provided in the `Name and Description` section described above. Any markdown-formatted text between the end of the initial function description and the beginning of the parameter list will be treated as additional explanatory notes, and will be read and stored separately from the initial description in the final data structure.
+
+Separating the main function description from this longer, more detailed explanation allows consumers who only need a condensed version of the docs (e.g. the InfluxDB UI) to do so, while other consumers are free to use the full docs.
+
+This section may be omitted.
+
+### Parameters
+
+Parameters should take the form of a markdown unordered list. Each list item should start with the name of the parameter wrapped in backticks, followed by a brief, one-line description.
+
+The parameters in the list should be ordered exactly as they are in the function signature, and no parameter should be omitted.
+
+Each list item should read as a complete sentence.
+
+_Example:_
+
+```
+// - `tables` is a stream of tables
+// - `method` is the method to use when joining
+// - `on` is a list of column names on which to join
+```
+
+### Code Examples
+
+Code examples consist of a heading, an optional preamble, and a markdown-formatted code block.
```

Comment:
Can you clarify where in the flux file the code examples go?

Reply:
I say at the top of the file that each section should go in the order presented in the document, but I can make that more explicit. Something like "Code examples should come immediately after the parameter list." Would that work?

Reply:
Yes the extra clarification is good.